src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneSelector.cs

```diff
@@ -27,10 +27,12 @@ public async Task<TimeZoneSelectorResult> GetTimeZoneAsync()
     {
         var currentUser = await _userManager.GetUserAsync(_httpContextAccessor.HttpContext.User);
 
-        return new TimeZoneSelectorResult
-        {
-            Priority = 100,
-            TimeZoneId = async () => (await _userTimeZoneService.GetAsync(currentUser)).TimeZoneId
-        };
+        return currentUser == null
+            ? null
+            : new TimeZoneSelectorResult
+            {
+                Priority = 100,
+                TimeZoneId = async () => (await _userTimeZoneService.GetAsync(currentUser))?.TimeZoneId
```

Comment:
```suggestion
                TimeZoneId = () => (_userTimeZoneService.GetAsync(currentUser))?.TimeZoneId
```

Reply:
@Piedone I don't think there is a need for the async here. just return a task would be better